main/core/src/mill/eval/Evaluator.scala

```diff
@@ -670,12 +670,16 @@ object Evaluator {
 
     val sortedGroups = Graph.groupAroundImportantTargets(topoSorted) {
       case t: NamedTask[Any] =>
-        val segments = t.ctx.segments
-
-        val additional =
-          if (!overriden(t)) Nil
-          else
-            Seq(Segment.Label("overriden")) ++ t.ctx.enclosing.split("\\.|#| ").map(Segment.Label)
+        val segments0 = t.ctx.segments
+        val (segments, additional) =
+          if (!overriden(t)) (segments0, Nil)
+          else {
+            val Segment.Label(tName) = segments0.value.last
+            (
+              Segments(segments0.value.reverse.tail.reverse: _*),
```

Comment:
`.reverse.tail.reverse` is just `.init`

Reply:
Wow, I'm using Scala for more than 10 years; today I learned something new.